aten/src/ATen/native/ForeachUtils.h

```diff
@@ -188,12 +201,66 @@ bool can_use_fast_route(TensorList tensors1, TensorList tensors2, TensorList ten
       return false;
     }
 
+    if (tensors1[i].strides() != tensors2[i].strides() ||
```

Comment:
There's a lot of repetition between can_use_fast_route(TL, TL), can_use_fast_route(TL, TL, TL), and can_use_fast_route(TL, TL, TL, Scalar) to check that "each respective Tensor has the same attributes" (for some definition of same attributes).

Would a helper function `has_same_attributes(TensorList tensors)` help? Then we could implement
```
bool can_use_fast_route(TensorList first, TensorList other) {
   for (int64_t i = 0; ...) {
       if not has_same_attributes({first[i], other[i]}) {
           return false;
       }
   }
   return true;
}

bool can_use_fast_route(TensorList first, TensorList second, TensorList third,) {
   for (int64_t i = 0; ...) {
       if not has_same_attributes({first[i], second[i], third[i]}) {
           return false;
       }
   }
   return true;
}

bool can_use_fast_route(TensorList first, TensorList second, TensorList third, Scalar fourth) {
   for (int64_t i = 0; ...) {
       if not has_same_attributes({first[i], second[i], third[i]}) {
           return false;
       }
       // check type promotion with Scalar fourth
       ...
   }
   return true;
}
```